source/php/App.php

```diff
@@ -120,57 +130,76 @@ public function hijackLogin($username)
-                            exit;
-                        }
-                        wp_redirect(apply_filters('adApiWpIntegration/login/subscriberRedirect', home_url($referer)));
+
+        //Only assume thre's a existing user id if autocreate is off
+        if (!defined('AD_AUTOCREATE_USER') || (defined('AD_AUTOCREATE_USER') && AD_AUTOCREATE_USER === false)) {
+            $this->userId = $this->getUserID($username);
+
+            if (is_numeric($this->userId) && !empty($this->userId)) {
+                return;
+            }
+        }
+
+        //Init required classes
+        $this->curl = new Helper\Curl();
+        $this->profile = new Profile();
+
+        //Fetch user from api
+        $result = $this->fetchUser($this->username, $this->password);
+
+        //Abort if theres a error
+        if($result !== false) {
+                
+            //Validate signon
+            if ($this->validateLogin($result, $this->username) && $result !== false) {
+                
+                //Auto create a user account
+                if (defined('AD_AUTOCREATE_USER') && AD_AUTOCREATE_USER === true) {
+                    Helper\AutoCreate::autoCreateUser($this->username, $this->password, $result);
+                }
+
+                //Get the user id
+                $this->userId = $this->getUserID($username);
+                
+                //Update user profile
+                $this->profile->update($result, $this->userId);
+
+                //Signon
+                $this->signOn(array(
+                    'user_login' => $this->username,
+                    'user_password' => $this->password,
+                    'remember' => isset($_POST['rememberme']) && $_POST['rememberme'] == "forever" ? true : false
+                ));
+
+                //Redirect to admin panel / frontpage
+                if (in_array('subscriber', (array) get_userdata($this->userId)->roles)) {
+
+                    //Get bulitin referer
+                    if (isset($_POST['_wp_http_referer'])) {
+                        $referer = $_POST['_wp_http_referer'];
+                    } else {
+                        $referer = "/";
+                    }
+
+                    //Redirect to correct url
+                    if (is_multisite()) {
+                        wp_redirect(apply_filters('adApiWpIntegration/login/subscriberRedirect', network_home_url($referer)));
                         exit;
                     }
+                    wp_redirect(apply_filters('adApiWpIntegration/login/subscriberRedirect', home_url($referer)));
+                    exit;
+                }
 
-                    wp_redirect(apply_filters('adApiWpIntegration/login/defaultRedirect', admin_url("?auth=active-directory")));
+                if (defined('AD_AUTOCREATE_USER') || AD_AUTOCREATE_USER === true) {
+                    wp_redirect(apply_filters('adApiWpIntegration/login/editorRedirect', home_url($referer)));
```

Comment:
This row seems quite odd, why would we need a special case for this? How is it related to editor redirect when ad auto create user is enabled?

Reply:
Hmm, not sure. I think is just needed a filter to redirect the user to the correct page after signing in. I just used that filter for convenience, but perhaps I should use the 'defaultRedirect' option below instead? Since it should always redirect every logged in user to the same place...

This is how I'm using the filter in my theme php file. 

`
add_filter('adApiWpIntegration/login/editorRedirect', array($this, 'ad_redirect'));
  
  public function ad_redirect($redirectUrl)
    {
        return $redirectUrl . '/protocol';
    }
`

Should I use defaultRedirect instead?

Reply:
Yes, i believe that is more suitable.

Reply:
Do we still need the if statement around the wp_redirect then?

I noticed now that if the other redirects would trigger before the default one then my redirect won't work. So then I need to use each filter (subscriberRedirect, editorRedirect and defaultRedirect) in my php file?